code/modules/events/tear.dm

```diff
@@ -1,47 +1,101 @@
+// Dimensional Tear - A rift appears randomly on the station, does the following:
+// Flickers nearby computers as an early warning.
+// After a few seconds, it breaks nearby computers and mirrors.
+// A portal appears, then it spawns a few hostile mobs, and a leader one.
+// Then the portal deletes itself.
+
+// Event setup
 /datum/event/tear
-	startWhen = 3
-	announceWhen = 20
-	endWhen = 50
+	name = "dimensional tear"
+	announceWhen = 6
+	endWhen = 10
 	var/obj/effect/tear/TE
 
-/datum/event/tear/announce()
-	GLOB.event_announcement.Announce("A tear in the fabric of space and time has opened. Expected location: [impact_area.name].", "Anomaly Alert", 'sound/AI/anomaly.ogg')
+/datum/event/tear/setup()
+	impact_area = findEventArea()
 
 /datum/event/tear/start()
 	var/turf/T = pick(get_area_turfs(impact_area))
 	if(T)
-		TE = new /obj/effect/tear(T.loc)
+		// Give ghosts some time to jump there before it begins.
+		var/image/alert_overlay = image('icons/mob/animal.dmi', "hellhound")
+		notify_ghosts("\A [src] is about to open in [get_area(T)].", title = "Dimensional Rift", source = T, alert_overlay = alert_overlay, action = NOTIFY_FOLLOW)
+		addtimer(CALLBACK(src, .proc/spawn_tear, T), 4 SECONDS)
 
-/datum/event/tear/setup()
-	impact_area = findEventArea()
+		// Energy overload; we flicker the computer screens as an early warning and for extra spookiness.
+		for(var/obj/machinery/computer/C in range(8, T))
+			C.flicker()
+
+/datum/event/tear/proc/spawn_tear(location)
+	TE = new /obj/effect/tear(location)
+
+/datum/event/tear/announce()
+	GLOB.event_announcement.Announce("A tear in the fabric of space and time has opened. Expected location: [impact_area.name].", "Anomaly Alert", 'sound/AI/anomaly.ogg')
 
 /datum/event/tear/end()
 	if(TE)
 		qdel(TE)
 
+// The portal
 /obj/effect/tear
-	name="Dimensional Tear"
-	desc="A tear in the dimensional fabric of space and time."
-	icon='icons/effects/tear.dmi'
-	icon_state="tear"
+	name = "dimensional tear"
+	desc = "A tear in the dimensional fabric of space and time."
+	icon = 'icons/effects/tear.dmi'
+	icon_state = "tear"
 	density = 0
 	anchored = 1
```

Comment:
```suggestion
```
Effects have no density and are anchored by default.

Reply:
done in 7ad6cfa0fa04efb8038a83189fe47d49e885ad27
I did not even notice this old code, thank you!